src/ChargeDrift/ChargeDrift.jl

```diff
@@ -80,12 +80,10 @@ function modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv
 end
 modulate_driftvector(sv::CartesianVector{T}, pt::CartesianPoint{T}, vdv::Missing) where {T} = sv
 
-@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cylindrical})::Bool where {T <: SSDFloat}
-    pt_cyl::CylindricalPoint{T} = CylindricalPoint(pt)
-    pt_cyl in point_types || pt_cyl in det.semiconductor
-end
-@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, Cartesian})::Bool where {T <: SSDFloat}
-    pt in point_types || pt in det.semiconductor
+@inline function _is_next_point_in_det(pt::CartesianPoint{T}, det::SolidStateDetector{T}, point_types::PointTypes{T, 3, S})::Bool where {T <: SSDFloat,S}
+    cpt = _convert_point(pt, S)
+    point_type::PointType = point_types[cpt]
+    (point_type & bulk_bit > 0)

```

Comment:
These three lines is exactly what `pt in point_types` does, see [PointTypes.jl#L63](https://github.com/SebastianRuffert/SolidStateDetectors.jl/blob/b013406e4e047f3d2f5791f2f77e9e94a29e5a9e/src/ScalarPotentials/PointTypes.jl#L63).

Also, if the `pt` does not have the `bulk_bit` set, we still might want to check the geometries, i.e. that the point is in the semiconductor, but **not** in the contacts, something like  `pt in det.semiconductor && !(pt in det.contacts)`.
Maybe we can do something with `CSGDifference`:
 `pt in sum(broadcast(s -> s.geometry, det.semiconductors)) - sum(broadcast(c -> c.geometry, det.contacts))`